src/wallet3/wallet.cpp

```diff
@@ -12,28 +12,32 @@
 #include <sqlitedb/database.hpp>
 #include <oxenmq/oxenmq.h>
 
-#include "common/fs.h"
 #include <future>
 #include <chrono>
 #include <thread>
 
 #include <iostream>
 
+#include <oxen/log.hpp>
+
+#include <spdlog/sinks/rotating_file_sink.h>
+
 namespace wallet
 {
+  static auto logcat = oxen::log::Cat("wallet");
+
   Wallet::Wallet(
       std::shared_ptr<oxenmq::OxenMQ> omq,
-      std::shared_ptr<Keyring> keys,
+      std::shared_ptr<Keyring> keyring,
       std::shared_ptr<TransactionConstructor> tx_constructor,
       std::shared_ptr<DaemonComms> daemon_comms,
       std::string_view dbFilename,
       std::string_view dbPassword,
       wallet::Config config_in)
       : omq(omq)
-      , db{std::make_shared<WalletDB>(fs::path(dbFilename), dbPassword)}
-      , keys{keys}
```

Comment:
Any reason not to keep the initialization of `keys` here?  Right now it's being done later with `if (keyring)` but if we just initialized `keys{std::move(keyring)},` here then the later code gets a bit simpler by just using `keys` and forgetting that `keyring` exists at all.

Reply:
passing a keyring here is when creating a wallet from a seed phrase, if you are opening an existing wallet file then keyring will be nullptr

Reply:
That's fine, you can still move an empty shared_ptr into a shared_ptr.